packages/main/src/plugin/kubernetes/health-checker.ts

```diff
@@ -0,0 +1,66 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { KubeConfig } from '@kubernetes/client-node';
+import { AbortError, Health } from '@kubernetes/client-node';
+
+export interface CheckOptions {
+  // timeout in ms
+  timeout?: number;
+}
+
+type ReadinessCallback = (ready: boolean) => void;
+
+// HealthChecker checks the readiness of a Kubernetes context
+// by requesting the readiness endpoint of its server
+export class HealthChecker {
```

Comment:
We should make it extend Disposable, so we can recycle all of that easily (abort)